### DISCUSSION TITLE
Failed to parse content of Authorization header

### DISCUSSION THREAD

**Discussion by mczapala** [authelia/authelia#7472] [Troubleshooting] [answered]
Hello, I recently updated authelia to new version that uses the new authz endpoints for authentication. From that point I cannot login to the jellyfin app because authelia responds with 401 error. This message can be seen in the logs:
```
time="2024-06-30T20:51:05+02:00" level=debug msg="Check authorization of subject username= groups= ip=192.168.88.26 and object https://jellyfin.domain.tld/Users/authenticatebyname (method POST)."
time="2024-06-30T20:51:05+02:00" level=debug msg="Error occurred while attempting to authenticate a request but the matched rule was a bypass rule" error="failed to parse content of Authorization header: invalid scheme: scheme with name 'mediabrowser' is unknown" method=GET path=/api/authz/forward-auth remote_ip=192.168.88.26
time="2024-06-30T20:51:05+02:00" level=debug msg="Responding 401 WWW-Authenticate" method=GET path=/api/authz/forward-auth remote_ip=192.168.88.26
```

The usecase looks like this:
1. I open the jellyfin page in the browser
2. I am prompted to sign in with authelia as per one_factor policy in the configuration
3. Authelia sign in works fine and now I am redirected to jellyfin sign in page
4. Some endpoints are returning 401 with the above lines appearing in the authelia logs

![image](https://github.com/authelia/authelia/assets/36843579/c9fa085a-6b24-4f19-92a9-d542f8fdd377)

Setting the policy to bypass resolves this issue, but I feel like this might be a bug, it was working with the api/verify method before.
Middleware config from traefik:
![image](https://github.com/authelia/authelia/assets/36843579/7ccf1e92-8a66-444b-9ab0-6c66b3d2a334)

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

It seems like the issue you're encountering is related to the new `authz` endpoints introduced in the latest version of Authelia. The error message indicates that the `Authorization` header contains an unknown scheme (`mediabrowser`), which is causing the authentication to fail.

Here are a few steps you can take to troubleshoot and potentially resolve this issue:

1. **Check Middleware Configuration**: Ensure that your Traefik middleware configuration is correctly set up to use the new `authz` endpoints. The configuration should look something like this:
   ```yaml
   - "traefik.http.middlewares.authelia.forwardauth.address=http://authelia:9091/api/authz/forward-auth"
   - "traefik.http.middlewares.authelia.forwardauth.trustForwardHeader=true"
   - "traefik.http.middlewares.authelia.forwardauth.authResponseHeaders=Remote-User,Remote-Groups,Remote-Name,Remote-Email"
   ```

2. **Authorization Header Scheme**: The error message suggests that the `Authorization` header scheme `mediabrowser` is unknown to Authelia. You might need to configure Authelia to recognize this scheme or modify the way Jellyfin sends the `Authorization` header.

3. **Bypass Rule**: As a temporary workaround, you mentioned that setting the policy to bypass resolves the issue. While this is not ideal for security, it can help you identify if the problem is strictly related to the new endpoint handling.

4. **Consult Documentation**: Refer to the [Integration Guide](https://www.authelia.com/integration/proxies/introduction) for detailed instructions on configuring the new `authz` endpoints. Ensure that all configurations are up-to-date with the latest changes.

5. **Community and Support**: If the issue persists, consider reaching out to the Authelia community or support channels. You can provide them with the detailed logs and configuration snippets to get more specific assistance.

Here are some relevant sources that might help you further:
- [Integration Guide](https://www.authelia.com/integration/proxies/introduction)
- [Server Endpoints Authz Configuration](https://www.authelia.com/configuration/miscellaneous/server-endpoints-authz)

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment, marked as answer]
The new authz endpoints are highly customizable but by default look for the Authorization header if present before checking cookies. You can adjust this to operate similar to the old endpoint however:

```yaml
server:
  endpoints:
    authz:
      forward-auth:
        implementation: 'ForwardAuth'
        authn_strategies:
          - name: 'HeaderProxyAuthorization'
            schemes:
              - 'Basic'
          - name: 'CookieSession'
```

**Reply by mczapala** [reply]
Works like a charm, thank you very much!

**Reply by Appoxo** [reply]
Works!
Tyvm
